<?xml version="1.0" encoding="utf-8" ?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot name="sampling_rover_with_arm_urdf">
    <link name="world" />
    <joint name="base_rover" type="fixed">
        <parent link="world" />
        <child link="rover" />
        <origin rpy="0 0 0" xyz="0.0 0.0 0.00001" />
    </joint>
    <link name="rover">
        <inertial>
            <origin xyz="0.04233 -0.030066 0.22201" rpy="0 0 0" />
            <mass value="23.83" />
            <inertia ixx="0.13598" ixy="0.0004091" ixz="0.0022915" iyy="0.10672" iyz="-0.0028185" izz="0.14333" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/rover.STL" />
            </geometry>
            <material name="">
                <color rgba="0.79216 0.81961 0.93333 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/rover.STL" />
            </geometry>
        </collision>
    </link>
    <link name="base_arm">
        <inertial>
            <origin xyz="7.5146E-16 1.2768E-15 0.062877" rpy="0 0 0" />
            <mass value="2.1128" />
            <inertia ixx="0.0074591" ixy="-6.4985E-07" ixz="2.9585E-17" iyy="0.0074553" iyz="-7.5664E-18" izz="0.010214" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/base_arm.STL" />
            </geometry>
            <material name="">
                <color rgba="0.79216 0.81961 0.93333 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/base_arm.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="base_joint" type="fixed">
        <origin xyz="-0.00070858 0.2487 0.409" rpy="0 0 0" />
        <parent link="rover" />
        <child link="base_arm" />
        <axis xyz="0 0 0" />
        <limit lower="0" upper="0" effort="0" velocity="0" />
    </joint>
    <link name="link_1">
        <inertial>
            <origin xyz="0.00064135 -0.0059228 0.051415" rpy="0 0 0" />
            <mass value="0.49072" />
            <inertia ixx="0.001755" ixy="3.1435E-07" ixz="-7.1879E-07" iyy="0.0010881" iyz="0.00019759" izz="0.001162" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/link_1.STL" />
            </geometry>
            <material name="">
                <color rgba="0.89804 0.91765 0.92941 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/link_1.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint_1" type="revolute">
        <origin xyz="0 0 0.131" rpy="0 0 0" />
        <parent link="base_arm" />
        <child link="link_1" />
        <axis xyz="0 0 1" />
        <limit lower="-3.142" upper="3.142" effort="100" velocity="2" />
    </joint>
    <link name="link_2">
        <inertial>
            <origin xyz="-3.148E-06 0.2843 0.0099169" rpy="0 0 0" />
            <mass value="1.1101" />
            <inertia ixx="0.0026516" ixy="-1.2212E-07" ixz="-1.1353E-07" iyy="0.0012227" iyz="-0.00020306" izz="0.0025953" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/link_2.STL" />
            </geometry>
            <material name="">
                <color rgba="0.79216 0.81961 0.93333 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/link_2.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint_2" type="revolute">
        <origin xyz="0 -0.0355 0.1519" rpy="1.5708 0 -1.5708" />
        <parent link="link_1" />
        <child link="link_2" />
        <axis xyz="0 0 1" />
        <limit lower="-3.142" upper="3.142" effort="100" velocity="2" />
    </joint>
    <link name="link_3">
        <inertial>
            <origin xyz="2.2204E-16 0.0011125 -0.0037525" rpy="0 0 0" />
            <mass value="0.83444" />
            <inertia ixx="0.0013519" ixy="9.0993E-07" ixz="9.3162E-20" iyy="0.0013665" iyz="-1.2183E-05" izz="0.0015146" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/link_3.STL" />
            </geometry>
            <material name="">
                <color rgba="0.79216 0.81961 0.93333 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/link_3.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint_3" type="revolute">
        <origin xyz="0 0.363 0.149" rpy="0 0 1.5708" />
        <parent link="link_2" />
        <child link="link_3" />
        <axis xyz="0 0 -1" />
        <limit lower="-3.142" upper="3.142" effort="50" velocity="4" />
    </joint>
    <link name="link_4">
        <inertial>
            <origin xyz="-0.0011125 -1.0547E-15 -0.0037525" rpy="0 0 0" />
            <mass value="0.83444" />
            <inertia ixx="0.0013665" ixy="-9.0993E-07" ixz="1.2183E-05" iyy="0.0013519" iyz="1.3053E-20" izz="0.0015146" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/link_4.STL" />
            </geometry>
            <material name="">
                <color rgba="0.79216 0.81961 0.93333 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/link_4.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint_4" type="revolute">
        <origin xyz="0 0.13161 0.10761" rpy="-1.5708 0 0" />
        <parent link="link_3" />
        <child link="link_4" />
        <axis xyz="0 0 -1" />
        <limit lower="-3.142" upper="3.142" effort="50" velocity="4" />
    </joint>
    <link name="link_5">
        <inertial>
            <origin xyz="0.0056805 -0.00015103 -0.30906" rpy="0 0 0" />
            <mass value="1.0858" />
            <inertia ixx="0.001469" ixy="-1.5038E-06" ixz="7.5505E-06" iyy="0.0014597" iyz="-1.0158E-06" izz="0.002048" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/link_5.STL" />
            </geometry>
            <material name="">
                <color rgba="0.79216 0.81961 0.93333 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/link_5.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint_5" type="revolute">
        <origin xyz="-0.45203 0 0.10768" rpy="-1.5708 0 1.5708" />
        <parent link="link_4" />
        <child link="link_5" />
        <axis xyz="0 0 -1" />
        <limit lower="-3.142" upper="3.142" effort="50" velocity="4" />
    </joint>
    <link name="Right_ee">
        <inertial>
            <origin xyz="0.011076 -0.16984 -0.00036022" rpy="0 0 0" />
            <mass value="0.15605" />
            <inertia ixx="0.0007532" ixy="1.5845E-05" ixz="-1.1744E-13" iyy="0.00044526" iyz="-2.7993E-13" izz="0.0004941" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/Right_ee.STL" />
            </geometry>
            <material name="">
                <color rgba="0.79216 0.81961 0.93333 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/Right_ee.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint_ree" type="revolute">
        <origin xyz="0 0.03156 -0.18278" rpy="-1.5708 0 1.5708" />
        <parent link="link_5" />
        <child link="Right_ee" />
        <axis xyz="0 0 1" />
        <limit lower="0" upper="0.8" effort="15" velocity="0" />
    </joint>
    <link name="Left_ee">
        <inertial>
            <origin xyz="-0.011076 -0.16984 -0.00041022" rpy="0 0 0" />
            <mass value="0.15605" />
            <inertia ixx="0.0007532" ixy="-1.5845E-05" ixz="-1.1743E-13" iyy="0.00044526" iyz="2.7992E-13" izz="0.0004941" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/Left_ee.STL" />
            </geometry>
            <material name="">
                <color rgba="0.1098 0.1098 0.1098 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://sampling_rover_with_arm_urdf/meshes/Left_ee.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint_lee" type="revolute">
        <origin xyz="0 -0.031415 -0.18278" rpy="-1.5708 0 1.5708" />
        <parent link="link_5" />
        <child link="Left_ee" />
        <axis xyz="0 0 1" />
        <limit lower="-0.8" upper="0" effort="15" velocity="6" />
        <mimic joint="joint_ree" multiplier="-1.0" offset="0.0" />
    </joint>
    <transmission name="link_1_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_1">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="link_1_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="link_2_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_2">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="link_2_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="link_3_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_3">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="link_3_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="link_4_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_4">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="link_4_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="link_5_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_5">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="link_5_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="Right_ee_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_ree">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="Right_ee_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <gazebo>
        <plugin name="control" filename="libgazebo_ros_control.so">
            <robotNamespace>/</robotNamespace>
        </plugin>
        <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
            <robotNamespace>/</robotNamespace>
        </plugin>
    </gazebo>
    <gazebo reference="rover">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="base_arm">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="link_1">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="link_2">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="link_3">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="link_4">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="link_5">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="Right_ee">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="Left_ee">
        <selfCollide>true</selfCollide>
    </gazebo>
    <!--  
  <gazebo>
    <plugin name="gazebo_grasp_fix" filename="libgazebo_grasp_fix.so">
      <arm>
        <arm_name>/</arm_name>
        <palm_link>link_5</palm_link>
        <gripper_link>Right_ee</gripper_link>
        <gripper_link>Left_ee</gripper_link>
      </arm>
      <forces_angle_tolerance>90</forces_angle_tolerance>
      <update_rate>10</update_rate>
      <grip_count_threshold>3</grip_count_threshold>
      <max_grip_count>10</max_grip_count>
      <release_tolerance>0.004</release_tolerance>
      <disable_collisions_on_attach>true</disable_collisions_on_attach>
      <contact_topic>__default_topic__</contact_topic>
    </plugin> 
  </gazebo>
  -->
    <transmission name="trans_joint_lee">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_lee">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_lee_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
</robot>

